tests/draft2019-09/ref.json

```diff
@@ -358,6 +358,216 @@
             }
         ]
     },
+    {
+        "description": "Location-independent identifier name, fragments disallowed",
+        "schema": {
+            "$ref": "#foo",
+            "$defs": {
+                "A": {
+                    "$id": "#foo",
```

Comment:
Isn't this an invalid schema now, not an invalid instance?

Reply:
Yes, but isn't everything under "$defs" considered a schema?
[8.2.5. Schema Re-Use With "$defs"](https://json-schema.org/draft/2019-09/json-schema-core.html#rfc.section.8.2.5)

Reply:
Right -- which is my point -- the thing that is in the test's "schema" should never be invalid -- users of the test suite should trust it to be itself valid, the point is to test the cases, not the schema of the test itself -- it seems like this should be written as a metaschema test -- where the schema is the metaschema (via just {`"ref": "$2019schema"`), and this kind of schema is the instance for it

Reply:
I see what you're saying. This schema should be in the "data" section, and the meta-schema should be in the "schema" section. Agreed; I'll fix this up.

Reply:
@Julian what do you think of this: https://github.com/json-schema-org/JSON-Schema-Test-Suite/issues/299#issuecomment-605697434